stl/inc/ranges

```diff
@@ -4748,6 +4748,566 @@ namespace ranges {
+                _RANGES advance(*_Parent->_Current, _Parent->_Remainder, _RANGES end(_Parent->_Range));
+                _Parent->_Remainder = _Parent->_Count;
+            }
+
+            _NODISCARD_FRIEND constexpr bool operator==(const _Outer_iterator& _Left, default_sentinel_t) noexcept(
+                noexcept(_Left._Is_end())) /* strengthened */ {
+                return _Left._Is_end();
+            }
+
+            _NODISCARD_FRIEND constexpr difference_type operator-(default_sentinel_t,
+                const _Outer_iterator& _Right) noexcept(noexcept(_Right._Get_size())) /* strengthened */
+                requires sized_sentinel_for<sentinel_t<_Vw>, iterator_t<_Vw>> {
+                return _Right._Get_size();
+            }
+
+            _NODISCARD_FRIEND constexpr difference_type operator-(const _Outer_iterator& _Left,
+                default_sentinel_t) noexcept(noexcept(_Left._Get_size())) /* strengthened */
+                requires sized_sentinel_for<sentinel_t<_Vw>, iterator_t<_Vw>> {
+                return -_Left._Get_size();
+            }
+        };
+
+    public:
+        // clang-format off
+        chunk_view() requires default_initializable<_Vw> = default;
+        // clang-format on
+
+        constexpr explicit chunk_view(_Vw _Range_, const range_difference_t<_Vw> _Count_) noexcept(
+            is_nothrow_move_constructible_v<_Vw>) /* strengthened */
+            : _Range(_STD move(_Range_)), _Count{_Count_} {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Count > 0, "n must be greater than 0");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        _NODISCARD constexpr _Vw base() const& noexcept(is_nothrow_copy_constructible_v<_Vw>) /* strengthened */
+            requires copy_constructible<_Vw> {
+            return _Range;
+        }
+
+        _NODISCARD constexpr _Vw base() && noexcept(is_nothrow_move_constructible_v<_Vw>) /* strengthened */ {
+            return _STD move(_Range);
+        }
+
+        _NODISCARD constexpr auto begin() noexcept(
+            noexcept(_RANGES begin(_Range)) && is_nothrow_move_constructible_v<iterator_t<_Vw>>) /* strengthened */ {
+            _Current._Emplace(_RANGES begin(_Range));
+            _Remainder = _Count;
+            return _Outer_iterator{*this};
+        }
+
+        _NODISCARD constexpr auto end() noexcept /* strengthened */ {
+            return default_sentinel;
+        }
+
+        _NODISCARD constexpr auto size() noexcept(noexcept(_RANGES distance(_Range))) /* strengthened */
+            requires sized_range<_Vw> {
+            return _To_unsigned_like(_Div_ceil(_RANGES distance(_Range), _Count));
+        }
+
+        _NODISCARD constexpr auto size() const noexcept(noexcept(_RANGES distance(_Range))) /* strengthened */
+            requires sized_range<const _Vw> {
+            return _To_unsigned_like(_Div_ceil(_RANGES distance(_Range), _Count));
+        }
+    };
+
+    template <forward_range _Vw>
+        requires view<_Vw>

```

Comment:
Why swap `forward_range` and `view` here from the standard?

```suggestion
    template <view _Vw>
        requires forward_range<_Vw>
```

Reply:
While this is not consistent with the standard, it is consistent with how most other views are defined in this repository.

So I would rather not change it.

Reply:
I personally agree with Tim Song here, I prefer the specialization part being placed in the `requires` clause (noted in the discord), and I also additionally prefer following the standard here. However, I'm only weakly opinionated here so if you don't want to make this change just resolve this conversation.

Reply:
Fine by me, its not something I would restart a full test round though.

Lets keep this open and wait for the next round of comments / next merge